test/libdnf/base/test_loggers.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "test_loggers.hpp"
+
+#include "libdnf/base/log_router.hpp"
+#include "libdnf/base/memory_buffer_logger.hpp"
+#include "libdnf/base/stream_logger.hpp"
+
+#include <memory>
+#include <sstream>
+
+CPPUNIT_TEST_SUITE_REGISTRATION(LoggersTest);
+
+
+void LoggersTest::setUp() {}
+
+
+void LoggersTest::tearDown() {}
+
+
+// Test of logger infrastructure:
+// 1. Write messages into a circular MemoryBufferLogger instance.
+// 2. Create a LogRouter instance with two StreamLogger instances attached.
+// 3. Write messages from MemoryBufferLoger instance into LogRouter instance.
+// 4. Write aditional message into LogRouter instance.
+// 5. Check content of streams of both StreamLogger instances.
+void LoggersTest::test_loggers() {
+
+    time_t msg_time = 1582604702;  // Timestamp of the first message. "2020-02-25T04:25:02Z"
+    const pid_t pid = 25;          // Process identifier.
+
+    // Text that is expected in the logs.
+    const std::string LOG_OUT =
+R"**(2020-02-25T04:25:06Z [25] INFO Info message
```

Comment:
indent by 8 spaces to make clang-format happy

Reply:
This is a raw string literal. But it does not contain escape sequences so I will use unprefixed string literal with indent.